packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -392,7 +397,12 @@ export class RestApi extends Resource implements IRestApi {
         ...props.deployOptions
       });
 
-      new CfnOutput(this, 'Endpoint', { value: this.urlForPath() });
+      const customExportName = props.exportName === undefined ? false : props.exportName;
+      if (customExportName) {
+        new CfnOutput(this, 'Endpoint', { exportName: customExportName, value: this.urlForPath() });
```

Comment:
I believe you should be able to just assign `customExportName` to `exportName`. If it will be undefined, it will just not export anything.